integration-cli/docker_cli_events_test.go

```diff
@@ -686,3 +686,81 @@ func (s *DockerRegistrySuite) TestEventsImageFilterPush(c *check.C) {
 	}
 }
+
+func (s *DockerSuite) TestEventsOOMDisableFalse(c *check.C) {
+	testRequires(c, DaemonIsLinux)
+
+	dockerCmd(c, "run", "--name", "oomFalse", "-m", "4MB", "busybox", "sh", "-c", "x=a; while true; do x=$x$x$x$x; done")
+
+	out, _ := dockerCmd(c, "events", "--since=0", "-f", "container=oomFalse", fmt.Sprintf("--until=%d", daemonTime(c).Unix()))
+	events := strings.Split(strings.TrimSuffix(out, "\n"), "\n")
+	if len(events) < 6 {
+		c.Fatalf("Missing expected event")
+	}
+
+	createEvent := strings.Fields(events[len(events)-6])
+	attachEvent := strings.Fields(events[len(events)-5])
+	startEvent := strings.Fields(events[len(events)-4])
+	resizeEvent := strings.Fields(events[len(events)-3])
+	oomEvent := strings.Fields(events[len(events)-2])
+	dieEvent := strings.Fields(events[len(events)-1])
+	if createEvent[len(createEvent)-1] != "create" {
+		c.Fatalf("event should be create, not %#v", createEvent)
+	}
+	if attachEvent[len(attachEvent)-1] != "attach" {
+		c.Fatalf("event should be attach, not %#v", attachEvent)
+	}
+	if startEvent[len(startEvent)-1] != "start" {
+		c.Fatalf("event should be start, not %#v", startEvent)
+	}
+	if resizeEvent[len(resizeEvent)-1] != "resize" {
+		c.Fatalf("event should be resize, not %#v", resizeEvent)
+	}
+	if oomEvent[len(oomEvent)-1] != "oom" {
+		c.Fatalf("event should be oom, not %#v", oomEvent)
+	}
+	if dieEvent[len(dieEvent)-1] != "die" {
+		c.Fatalf("event should be die, not %#v", dieEvent)
+	}
+}
+
+func (s *DockerSuite) TestEventsOOMDisableTrue(c *check.C) {
+	testRequires(c, DaemonIsLinux)
+
+	dockerCmd(c, "run", "--oom-kill-disable=true", "--name", "oomTrue", "-m", "4MB", "busybox", "sh", "-c", "x=a; while true; do x=$x$x$x$x; done")
+
+	out, _ := dockerCmd(c, "events", "--since=0", "-f", "container=oomTrue", fmt.Sprintf("--until=%d", daemonTime(c).Unix()))
+	events := strings.Split(strings.TrimSuffix(out, "\n"), "\n")
+	if len(events) != 5 {
+		c.Fatalf("Missing expected event")
+	}
+
+	createEvent := strings.Fields(events[len(events)-5])
+	attachEvent := strings.Fields(events[len(events)-4])
+	startEvent := strings.Fields(events[len(events)-3])
+	resizeEvent := strings.Fields(events[len(events)-2])
+	oomEvent := strings.Fields(events[len(events)-1])
+
+	if createEvent[len(createEvent)-1] != "create" {
+		c.Fatalf("event should be create, not %#v", createEvent)
+	}
+	if attachEvent[len(attachEvent)-1] != "attach" {
+		c.Fatalf("event should be attach, not %#v", attachEvent)
+	}
+	if startEvent[len(startEvent)-1] != "start" {
+		c.Fatalf("event should be start, not %#v", startEvent)
+	}
+	if resizeEvent[len(resizeEvent)-1] != "resize" {
+		c.Fatalf("event should be resize, not %#v", resizeEvent)
+	}
+	if oomEvent[len(oomEvent)-1] != "oom" {
+		c.Fatalf("event should be oom, not %#v", oomEvent)
+	}
+
+	out, _ = dockerCmd(c, "inspect", "-f", "{{.State.Status}}", "oomTrue")
+	if out != "running" {
+		c.Fatalf("container should be still running, not %v", out)
+	}
+
+	dockerCmd(c, "stop", "oomTrue")
```

Comment:
This should just be `kill`, and probably as a defer after it's created. just in case something really goes wrong.